### PR TITLE
NavigationPerformanceEvent and NavigationPerformanceMetadata convert to kotlin

### PR DESCRIPTION
**Pull request by Lebedinsky**
## Description

Migrate to Kotlin [issue](https://github.com/mapbox/navigation-sdks/issues/166)

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationPerformanceMetadata.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    val os: String,
+    val gpu: String,
+    val manufacturer: String
+) {
+
+    companion object {
+        @JvmStatic
+        fun builder() = Builder()
+
+        class Builder {
```
Comment (RingerJK) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationPerformanceMetadata.kt]: private constructor

Comment (Lebedinsky) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationPerformanceMetadata.kt]: We can make it internal only.

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationPerformanceEvent.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        private const val DATE_AND_TIME_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+        private const val PERFORMANCE_TRACE = "mobile.performance_trace"
+        private const val EVENT_NAME = "event_name"
+        private val dateFormat = SimpleDateFormat(DATE_AND_TIME_PATTERN, Locale.US)
+    }
+
+    private val event: String
+    private val created: String
+    private val counters: MutableList<Counter<out Number>>
+    private val attributes: MutableList<Attribute>
```
Comment (RingerJK) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationPerformanceEvent.kt]: private val event: String = PERFORMANCE_TRACE
    private val created: String = obtainCurrentDate()
    private val counters: MutableList<Counter<out Number>> = ArrayList()
    private val attributes: MutableList<Attribute> = ArrayList()

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationPerformanceEvent.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import android.annotation.SuppressLint
+import android.os.Parcel
+import android.os.Parcelable
+import com.mapbox.android.telemetry.Event
+import java.text.SimpleDateFormat
+import java.util.ArrayList
+import java.util.Date
+import java.util.Locale
+
+@SuppressLint("ParcelCreator")
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationPerformanceEvent.kt]: `@SuppressWarnings("ParcelableCreator")` was removed, is it safe?

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationPerformanceMetadata.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    val brand: String,
+    val ram: String,
+    val os: String,
+    val gpu: String,
+    val manufacturer: String
+) {
+
+    companion object {
+        @JvmStatic
+        internal fun builder() = Builder()
```
Comment (RingerJK) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationPerformanceMetadata.kt]: looks like this `Metadata` is only for internal usage? may be better move class into `internal` package and use default values instead `Builder`

Comment (Lebedinsky) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationPerformanceMetadata.kt]: Added Fixme for class to mark it should be refactored as you described above when NavigationTelemetry and NavigationMetricsWrapper are migrated to Kotlin

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationPerformanceEvent.kt (hunk 3 above)
Comment (Lebedinsky) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationPerformanceEvent.kt]: It is not dangerous. Checked, lint is ok without this suppress.

**Pull request merged**